### DISCUSSION TITLE
getSession / getUser return null even after a successful setSession or refreshSession

### DISCUSSION THREAD

**Discussion by zhamid-src** [supabase/supabase#19608] [Questions]
I am using SSR and the documentation for javascript is sadly, incomplete. My use-case is the reset-password flow and I followed the instructions from https://supabase.com/docs/guides/auth/auth-password-reset#server-side-rendering-ssr

(Aside: the instructions in the above link contradict the instructions from https://supabase.com/docs/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr?framework=express as one redirects to auth/callback and the other to auth/confirm).

I kept getting "AuthSessionMissingError: Auth session missing" from updateUser from the new-password flow. So I decided to pass in the refresh_token and access_token. 

`const session = await supabase.auth.setSession({
      access_token,
      refresh_token,
    });
`

This works and I get a session with the user and all user details. But immediately after if I do:

`const user = await supabase.auth.getUser();
`
then I get a null user with this error:

`error: AuthApiError: invalid claim: missing sub claim`

If instead I call getSession, I get an error as well.

If I just call updateUser after setSession, I get:

`error: AuthSessionMissingError: Auth session missing!`

I can't figure out what I am doing wrong and the documentation (which I believe is supabase's weakpoint) isn't helpful.

**Comment by zhamid-src** [comment]
I've tried every possible thing. I followed the instructions to the T.  

Here's an overly simplified example:

`const authResponse: AuthResponse = await supabase.auth.verifyOtp({
          type,
          token_hash,
        });
        
       // ^ succeeds. I get the payload with the user and all tokens

       supabase.auth.updateUser({ password: "$R#W$R#$@$R@#$R@#$REWF" });`
       // ^ fails with "Auth session missing!"
`

If the user is logged in, then update User should work.

**Comment by tyforcode** [comment]
Also noting that I'm running into the same issue. Right after a successful login, I receive an error response calling `supabase.getUser()`
<img width="615" alt="Screenshot 2023-12-18 at 3 07 49 PM" src="https://github.com/supabase/supabase/assets/8219641/c4c4b114-c9c1-44c4-887f-c22f0ff4f647">
<img width="453" alt="Screenshot 2023-12-18 at 3 07 55 PM" src="https://github.com/supabase/supabase/assets/8219641/ea2f0838-61b0-44fb-8ba8-ce191a6dcff9">

The session also does not persist if I refresh the web page

**Comment by tyforcode** [comment]
@zhamid-src Ok, I found a solution. I had to fill in the `cookies` field in the supabase client so when I create it, it looks like:
```ts
import { createBrowserClient } from '@supabase/ssr'
import { Database } from '../models/database-schema'

const supabase = createBrowserClient<Database>(
  process.env.NEXT_PUBLIC_SUPABASE_URL!,
  process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
  {
    cookies: {
      get(name: string) {
        return localStorage.getItem(name)
      },
      set(name: string, value: string) {
        localStorage.setItem(name, value)
      },
      remove(name: string) {
        localStorage.removeItem(name)
      }
    }
  }
)
```

This resolved both of my issues above

**Reply by silentworks** [reply]
This is not a solution, this is just reverting to using normal `@supabase/supabase-js`. You should not do this if you are using the `@supabase/ssr` package. If you are using NextJS with only client side stuff then just use `@supabase/supabase-js`.

**Comment by zhamid-src** [comment]
I had a couple of other issues with supabase as well.  I ended up implementing my own auth. It took less time than I had spent on supabase. 

I think supabase is great if you want to use the client libraries, but the documentation is lacking if you want to keep auth on the server.

**Comment by karthikjn01** [comment]
Running into this issue also, for the time being I'll have to use the old way of doing things.
